python-libraries/data-platform-catalogue/data_platform_catalogue/client/datahub/datahub_client.py

```diff
@@ -231,8 +241,8 @@ def upsert_table(
             description=metadata.description,
             customProperties={
                 "sourceDatasetName": metadata.source_dataset_name,
-                "sourceDatasetLocation": metadata.source_dataset_location,
-                "sensitivityLevel": metadata.data_sensitivity_level,
+                "whereToAccessDataset": metadata.where_to_access_dataset,
+                "sensitivityLevel": str(metadata.data_sensitivity_level),
```

Comment:
There's an existing type incompatibility here as these are optional arguments (can be `None`), but `customProperties` is expecting `dict[str, str]`. 

We should change the metadata fields for use in the `customProperties` dict in `entities.py` from `str | None = None` to `str = ""`

Reply:
Done